src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp

```diff
@@ -332,21 +331,127 @@ void VTKPolyDataWriterInterface::writeElementFields( vtkSmartPointer< vtkCellDat
     }
   } );
 
-  for( auto const & field : allFields )
+  for( string const & field : allFields )
   {
-    vtkSmartPointer< VTKGEOSXData > data = VTKGEOSXData::New();
-    data->SetNumberOfValues( er.getNumberOfElements< SUBREGION >() );
-    data->SetName( field.c_str() );
 
-    localIndex count = 0;
+    // Here is why we need a special treatment for some fields here.
+    // - Fields that have 3 dimensions or less can be treated in a purely generic fashion using the writeField function
+    // - Some fields have more than 3 dimensions (cell id, plus 3 others) and cannot be easily visualized in Paraview.
+    //   For now, we apply a special component-by-component treatment to 4-dimension fields (see below), but we do not output fields with 5
+    // dimensions or more.
+
+    // First, check if the field requires a special treatment. For now, array4d< real64 > (for instance, for phaseCompFraction) are
+    // treated separately to make sure that they can be easily visualized in Paraview.
+    bool is4DField = false;
     er.forElementSubRegions< SUBREGION >( [&]( auto const & esr )
     {
       WrapperBase const & wrapper = esr.getWrapperBase( field );
-      writeField( wrapper, data, esr.size(), count );
+      std::type_info const & typeID = wrapper.getTypeId();
+      if( typeID==typeid(array4d< real64 >) )
+      {
+        is4DField = true;
+      }
     } );
```

Comment:
Is there a way to extract the dimension of the field instead? (like `array1d -> 1`, `array2d -> 2`...).
That way you could inform the user that there is a non supported type.

Reply:
I do not know if there is a way to extract the dimension of the field from the wrapper, other than using `typeid`. In the current code and in this PR, if the dimension of the field is not supported, an error message will be displayed and stop the simulation:
```
***** LOCATION: /g/g90/hamon1/geosx/GEOSX/src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp:442                                     
***** Controlling expression (should be false): true                                                                                           
***** ERROR                                                                                                                                    
***** LOCATION: /g/g90/hamon1/geosx/GEOSX/src/coreComponents/common/DataTypes.hpp:917                                                          
***** Controlling expression (should be false): true                                                                                           
***** Rank 0: TypeID not recognized. 
```
Even if it may not be very generic, I reused the logic of SiloFile.cpp, for instance here: 
https://github.com/GEOSX/GEOSX/blob/64bba253b61010d2810415c892b5227a2e3dc14f/src/coreComponents/fileIO/silo/SiloFile.cpp#L916-L921

Reply:
We're going to have to change it very soon (both in Silo and VTK outputs), because the `typeid` check does not work with layout permutations (and so will fail or work incorrectly on CUDA builds).

The easiest "generic" mechanics is to add a new methods to `WrapperBase` and `Wrapper`. Call it something like `numArrayDims()` and have it return NDIM for `LvArray::Array` and perhaps 0 for all other types (using template specialization in `wrapperHelpers`, similar to other specialized methods). The mechanism is a bit overused and keeps inflating the API surface of `Wrapper` class, but it's quite reliable.

Reply:
Thanks for the suggestion, I implemented what you suggested (but it is worth double checking what I have done :) )